scipy/stats/stats.py

```diff
@@ -2315,6 +2320,275 @@ def zmap(scores, compare, axis=0, ddof=0):
+    # consistent with `np.var` and `np.std`.
+    if not x.size:
+        return np.nan
+
+    # An error may be raised here, so fail-fast, before doing lengthy
+    # computations, even though `scale` is not used until later
+    if isinstance(scale, string_types):
+        scale = _iqr_scale_to_num(scale)
+
+    # Select the percentile function to use based on nans and policy
+    contains_nan, nan_policy = _contains_nan(x, nan_policy)
+
+    if contains_nan and nan_policy == 'omit':
+        percentile_func = _iqr_nanpercentile
+    else:
+        percentile_func = _iqr_percentile
+
+    if len(rng) != 2:
+        raise TypeError("quantile range must be two element sequence")
+
+    rng = sorted(rng)
+    pct = percentile_func(x, rng, axis=axis, interpolation=interpolation,
+                          keepdims=keepdims, contains_nan=contains_nan)
+    out = np.subtract(pct[1], pct[0])
+
+    if scale != 1.0:
+        out /= scale
+
+    return out
+
+
+def _iqr_percentile(x, q, axis=None, interpolation='linear', keepdims=False, contains_nan=False):
+    """
+    Private wrapper that works around older versions of `numpy`.
+
+    While this function is pretty much necessary for the moment, it
+    should be removed as soon as the minimum supported numpy version
+    allows.
+    """
+    if contains_nan and NumpyVersion(np.__version__) < '1.11.0':
+        # I see no way to avoid the version check to ensure that the corrected
+        # NaN behavior has been implemented except to call `percentile` on a
+        # small array.
+        msg = "Keyword nan_policy='propagate' not correctly supported for " \
+              "numpy versions < 1.11.0. The default behavior of " \
+              "`numpy.percentile` will be used."
+        warnings.warn(msg, RuntimeWarning)
+
+    try:
+        result = np.percentile(x, q, axis=axis, keepdims=keepdims,
+                               interpolation=interpolation)
+    except TypeError:
+        if interpolation != 'linear' or keepdims:
+            # At time or writing, this means np.__version__ < 1.9.0
+            warnings.warn("Keywords interpolation and keepdims not supported "
+                          "for your version of numpy", RuntimeWarning)
+        try:
+            # Special processing if axis is an iterable
+            original_size = len(axis)
+            axis = np.unique(axis)
+            if original_size > axis.size:
+                # mimic numpy if axes are duplicated
+                raise ValueError("duplicate value in axis")
+            if axis.size == x.ndim:
+                # axis includes all axes: revert to None
+                axis = None
+            elif axis.size == 1:
+                # no rolling necessary
+                axis = axis[0]
+            else:
+                # roll multiple axes to the end and flatten that part out
+                for ax in axis[::-1]:
+                    x = np.rollaxis(x, ax, x.ndim)
```

Comment:
To support multiple axes (without `keepdims`), I independently implemented a subset of `np.function_base._ureduce`. The main difference is that `_ureduce` uses `x.swapaxes` while I use `np.rollaxis`. The functionality is effectively the same since all the reduction axes get flattened out. Besides where the axes end up, is there any difference between the two methods (such as performance) that I should be aware of?